dojo_plugin/utils/awards.py

```diff
@@ -74,13 +60,26 @@ def get_belts():
     return result
 
 def update_awards(user):
-    current_belts = get_user_belts(user)
-    for belt in current_belts:
-        belt_award = Belts.query.filter_by(user=user, name=belt).first()
-        if belt_award:
+    current_belts = [belt.name for belt in Belts.query.filter_by(user=user)]
+    discord_user = get_discord_user(user.id)
+    discord_roles = get_discord_roles()
+    for belt, dojo_id in BELT_REQUIREMENTS.items():
+        belt_role = belt.title() + " Belt"
+        missing_role = discord_user and discord_roles.get(belt_role) not in discord_user["roles"]
+        if belt in current_belts and not missing_role:
             continue
-        db.session.add(Belts(user=user, name=belt))
-        db.session.commit()
+        dojo = Dojos.query.filter(Dojos.official, Dojos.id == dojo_id).first()
+        if not (dojo and dojo.completed(user)):
```

Comment:
This would redundantly fire even if a user already has the belt role, eating some small amounts of resources.

Reply:
Yeah, you're right; the existence of the belt award should definitely be saving us from doing this work.